packages/sdk38/src/lcdapi/lcdclient.ts

```diff
@@ -0,0 +1,331 @@
+  public static withExtensions<A extends object, B extends object, C extends object, D extends object>(
+    options: LcdClientBaseOptions,
+    setupExtensionA: LcdExtensionSetup<A>,
+    setupExtensionB: LcdExtensionSetup<B>,
+    setupExtensionC: LcdExtensionSetup<C>,
+    setupExtensionD: LcdExtensionSetup<D>,
+  ): LcdClient & A & B & C & D;
+
+  /** Constructs an LCD client with 5 extensions */
+  public static withExtensions<
+    A extends object,
+    B extends object,
+    C extends object,
+    D extends object,
+    E extends object
+  >(
+    options: LcdClientBaseOptions,
+    setupExtensionA: LcdExtensionSetup<A>,
+    setupExtensionB: LcdExtensionSetup<B>,
+    setupExtensionC: LcdExtensionSetup<C>,
+    setupExtensionD: LcdExtensionSetup<D>,
+    setupExtensionE: LcdExtensionSetup<E>,
+  ): LcdClient & A & B & C & D & E;
+
+  /** Constructs an LCD client with 6 extensions */
+  public static withExtensions<
+    A extends object,
+    B extends object,
+    C extends object,
+    D extends object,
+    E extends object,
+    F extends object
+  >(
+    options: LcdClientBaseOptions,
+    setupExtensionA: LcdExtensionSetup<A>,
+    setupExtensionB: LcdExtensionSetup<B>,
+    setupExtensionC: LcdExtensionSetup<C>,
+    setupExtensionD: LcdExtensionSetup<D>,
+    setupExtensionE: LcdExtensionSetup<E>,
+    setupExtensionF: LcdExtensionSetup<F>,
+  ): LcdClient & A & B & C & D & E & F;
+
+  /** Constructs an LCD client with 7 extensions */
+  public static withExtensions<
+    A extends object,
+    B extends object,
+    C extends object,
+    D extends object,
+    E extends object,
+    F extends object,
+    G extends object
+  >(
+    options: LcdClientBaseOptions,
+    setupExtensionA: LcdExtensionSetup<A>,
+    setupExtensionB: LcdExtensionSetup<B>,
+    setupExtensionC: LcdExtensionSetup<C>,
+    setupExtensionD: LcdExtensionSetup<D>,
+    setupExtensionE: LcdExtensionSetup<E>,
+    setupExtensionF: LcdExtensionSetup<F>,
+    setupExtensionG: LcdExtensionSetup<G>,
+  ): LcdClient & A & B & C & D & E & F & G;
+
+  /** Constructs an LCD client with 8 extensions */
+  public static withExtensions<
+    A extends object,
+    B extends object,
+    C extends object,
+    D extends object,
+    E extends object,
+    F extends object,
+    G extends object,
+    H extends object
+  >(
+    options: LcdClientBaseOptions,
+    setupExtensionA: LcdExtensionSetup<A>,
+    setupExtensionB: LcdExtensionSetup<B>,
+    setupExtensionC: LcdExtensionSetup<C>,
+    setupExtensionD: LcdExtensionSetup<D>,
+    setupExtensionE: LcdExtensionSetup<E>,
+    setupExtensionF: LcdExtensionSetup<F>,
+    setupExtensionG: LcdExtensionSetup<G>,
+    setupExtensionH: LcdExtensionSetup<H>,
+  ): LcdClient & A & B & C & D & E & F & G & H;
+
+  public static withExtensions<
+    A extends object,
+    B extends object,
+    C extends object,
+    D extends object,
+    E extends object,
+    F extends object,
+    G extends object,
+    H extends object
```

Comment:
Do we need this many though?

Reply:
Yes, it is a bit of a monster. But it is TypeScript overloading as documented.

Looking at https://github.com/CosmWasm/cosmjs/issues/276 there are 8 modules including CosmWasm. This can easily get more when custom chain modules come in. So I think the overloads are reasonable, even if we don't have examples that big.